cms/search/publishers.py

```diff
@@ -0,0 +1,202 @@
+            message.update(self._construct_release_specific_fields(page))
+        return message
+
+    def _construct_release_specific_fields(self, page: "Page") -> dict:
+        """Constructs and returns a dictionary with release-specific fields."""
+        release_fields = {
+            "finalised": page.status in ["CONFIRMED", "PROVISIONAL"],
+            "cancelled": page.status == "CANCELLED",
+            "published": page.status == "PUBLISHED",
+            "date_changes": [],
+        }
+
+        # If page.changes_to_release_date exists, add the first item's reason & previous date
+        if hasattr(page, "changes_to_release_date") and page.changes_to_release_date:
+            first_change = page.changes_to_release_date[0].value
+            release_fields["date_changes"].append(
+                {
+                    "change_notice": first_change.get("reason_for_change"),
+                    "previous_date": str(first_change.get("previous_date").isoformat().replace("+00:00", "Z"))
+                    if first_change.get("previous_date")
+                    else None,
+                }
+            )
+        return release_fields
+
+    def _construct_message_for_delete(self, page: "Page") -> dict:
+        """Build a dict that matches the 'content-deleted' schema:
+        - required: uri
+        - optional: trace_id.
+        """
+        message = {
+            "uri": page.url_path,
+        }
+
+        if hasattr(page, "trace_id"):
+            message["trace_id"] = page.trace_id
+
+        return message
+
+    def _map_page_type_to_content_type(self, page: "Page") -> Optional[str]:
+        """Maps the class name of a given page object to a corresponding content type string."""
+        mapping = {
+            "ReleaseCalendarPage": "release",
+            "StatisticalArticlePage": "bulletin",
+            "InformationPage": "static_page",
+            "IndexPage": "static_landing_page",
+            "MethodologyPage": "static_methodology",
+        }
+
+        return mapping.get(page.__class__.__name__)
+
+
+class KafkaPublisher(BasePublisher):
+    """Publishes messages to Kafka for 'search-content-updated' (created or updated)
+    and 'search-content-deleted' (deleted) events, aligning with the StandardPayload
+    / ReleasePayload / content-deleted schema definitions.
+    """
+
+    def __init__(self) -> None:
+        # Read Kafka configs settings
+        self.kafka_server = settings.KAFKA_SERVER
+        self._topic_created_or_updated: Optional[str] = settings.KAFKA_TOPIC_CREATED_OR_UPDATED
+        self._topic_deleted: Optional[str] = settings.KAFKA_TOPIC_DELETED
+
+        self.producer = KafkaProducer(
+            bootstrap_servers=[self.kafka_server],
+            api_version=(3, 8, 0),
+            value_serializer=lambda v: json.dumps(v).encode("utf-8"),
+        )
+
+    def get_topic_created_or_updated(self) -> str:
+        if self._topic_created_or_updated is None:
```

Comment:
This check and the check below should either happen in `__init__`, or even better as a system check https://docs.djangoproject.com/en/5.1/topics/checks/

Reply:
Done!